addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/CorsScanRule.java

```diff
@@ -0,0 +1,188 @@
+            "null",
+            handyScheme + RANDOM_NAME + "." + authority,
+            handyScheme + authority + "." + RANDOM_NAME + ".com",
+            // URL encoded backtick used to bypass weak Regex matching only alphanumeric chars to
+            // validate the domain: https://www.corben.io/tricky-CORS/
+            handyScheme + authority + "%60" + RANDOM_NAME + ".com",
+            null,
+            handyScheme + authority
+        };
+
+        boolean secScheme = false;
+        if (scheme == "https") {
+            secScheme = true;
+            payloads[5] = "http://" + authority;
+        }
+
+        for (String payload : payloads) {
+            HttpMessage msg = getNewMsg();
+            msg.getRequestHeader().setHeader(HttpRequestHeader.ORIGIN, payload);
+            try {
+                sendAndReceive(msg);
+
+                HttpResponseHeader respHead = msg.getResponseHeader();
+                String acaoVal = respHead.getHeader(HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN);
+
+                // If there is an ACAO header an alert will be triggered
+                if (acaoVal == null) {
+                    continue;
+                }
+
+                int risk = Alert.RISK_INFO;
+                boolean vuln = false;
+                String acacVal = respHead.getHeader(ACAC);
+                acacVal = acacVal == null ? "" : acacVal;
+
+                // Evaluates the risk for this alert
+                if (acaoVal.contains("*")) {
+                    vuln = true;
+                    risk = Alert.RISK_MEDIUM;
+                } else if (acaoVal.contains(RANDOM_NAME)
+                        || acaoVal.contains("null")
+                        || (secScheme && acaoVal.contains("http:"))) {
+                    vuln = true;
+                    // If authenticated AJAX requests are allowed, the risk is higher
+                    risk = (acacVal == "") ? Alert.RISK_MEDIUM : Alert.RISK_HIGH;
+                }
+                Matcher m =
+                        Pattern.compile(
+                                        String.format(
+                                                "^\\s*%s[:\\s]+%s(\\s+%s[:\\s]+%s)?",
+                                                HttpHeader.ACCESS_CONTROL_ALLOW_ORIGIN,
+                                                Pattern.quote(acaoVal),
+                                                ACAC,
+                                                acacVal),
+                                        Pattern.MULTILINE)
+                                .matcher(respHead.toString());
+                buildAlert(risk)
+                        .setMessage(msg)
+                        .setAttack(HttpRequestHeader.ORIGIN + ": " + payload)
+                        .setEvidence(m.find() ? m.group(0) : null)
+                        .raise();
+                return;
+            } catch (IOException e) {
+                LOG.error(e.getMessage(), e);
```

Comment:
This should be more a warn than error, this is expected to happen when there are connection problems (which is not an issue in ZAP).